man/bootctl.xml

```diff
@@ -277,6 +277,17 @@
         Partitions, or that multiple OS instances are not possible.</para></listitem>
       </varlistentry>
 
+      <varlistentry>
+        <term><option>-a</option></term>
+        <term><option>--arch=auto|all|ARCH</option></term>
+        <listitem><para>Install binaries for the specified EFI architecture (e.g. <literal>x64</literal> or
+        <literal>ia32</literal> on x86).</para>
+
+        <para>Defaults to <literal>auto</literal>, which will use the kernel-reported EFI platform size of
+        the running system to decide the right binaries to use. Setting this to <literal>all</literal> will
+        install binaries for all available architectures (this implies <option>--no-variables</option>).</para></listitem>
+      </varlistentry>
```

Comment:
so this feature as it looks right now seems to support exactly one or two archs, which are compiled into bootctl. 

I think we should make this entirely generic, so that we copy basically `/usr/lib/systemd/boot/efi/systemd-boot*.efi`, i.e. any arch that is supported. Or if you specifiy --arch= any subset of it.

i.e. I think distros should be able to implement multiarch to the max if they like and have arm and x86 EFI binaries of sd-boot in place if they like, and "bootctl" should be able to drop it in.

Or in otherwords: instead of compiled in list of archs, just pick up every file that is installed.

Reply:
> Or in otherwords: instead of compiled in list of archs, just pick up every file that is installed.

This is what the `all` arch is for. And I believe this should be explicitly opt-in. I certainly wouldn't want bootctl to always install ia32 binaries when I don't need them.

I now made it so that you can select from any efi arches we support, but we still default to auto detecting and only installing for that arch, which is imho the right thing to do.

Reply:
I'd argue here that if you have those other flavours installed, then we should install them too. i.e. I'd see that as a drop-in thing: if the other flavours are installed we take them too. Why would you install them in /usr/ if you wouldn't also want them in /efi/?